coalib/results/SourceRange.py

```diff
@@ -120,6 +120,38 @@ def expand(self, file_contents):
                                        tr.end.line,
                                        tr.end.column)
 
+    @enforce_signature
+    def affected_source(self, file_dict: dict):
+        r"""
+        Make an iterator that tells which are the lines affected in the
+        file within a specified range.
+
+        >>> from os.path import abspath
+        >>> sr = SourceRange.from_values('file_name', start_line=2, end_line=2)
+        >>> list(sr.affected_source({abspath('file_name'):
```

Comment:
When somebody passes the name of the file to `SourceRange.from_values`, it automatically converts the name of the file to absolute path. see `coalib.results.SourcePosition.py` constructor.